include/silo_api/database_mutex.h

```diff
@@ -0,0 +1,35 @@
+#ifndef SILO_DATABASE_MUTEX_H
+#define SILO_DATABASE_MUTEX_H
+
+#include <mutex>
+#include <shared_mutex>
+
+#include "silo/database.h"
+
+namespace silo_api {
+
+class FixedDatabase {
```

Comment:
I find the name a bit confusing. How about this?
```suggestion
class DatabaseLock {
```

Or `LockedDatabase`?